modules/qortexRtdProvider.js

```diff
@@ -144,22 +281,83 @@ export function loadScriptTag(config) {
  * @param {Object} config module config obtained during init
  */
 export function initializeModuleData(config) {
-  const DEFAULT_API_URL = 'https://demand.qortex.ai';
   const {apiUrl, groupId, bidders} = config.params;
-  requestUrl = `${apiUrl || DEFAULT_API_URL}/api/v1/analyze/${groupId}/prebid`;
-  bidderArray = bidders;
-  impressionIds = new Set();
-  currentSiteContext = null;
+  const qortexUrlBase = apiUrl || DEFAULT_API_URL;
+  const windowUrl = window.top.location.host;
+  qortexSessionInfo.bidderArray = bidders;
+  qortexSessionInfo.impressionIds = new Set();
+  qortexSessionInfo.currentSiteContext = null;
+  qortexSessionInfo.pageAnalysisdata = {
+    requestSuccessful: null,
+    analysisGenerated: false,
+    contextAdded: {}
+  };
+  qortexSessionInfo.sessionId = generateSessionId();
+  qortexSessionInfo.groupId = groupId;
+  qortexSessionInfo.groupConfigUrl = `${qortexUrlBase}/api/v1/prebid/group/configs/${groupId}/${windowUrl}`
+  qortexSessionInfo.contextUrl = `${qortexUrlBase}/api/v1/prebid/${groupId}/page/lookup`
+  qortexSessionInfo.pageAnalyisUrl = `${qortexUrlBase}/api/v1/prebid/${groupId}/page/index`;
+  qortexSessionInfo.analyticsUrl = generateAnalyticsHostUrl(qortexUrlBase);
+  return qortexSessionInfo;
+}
+
+export function saveContextAdded(reqBids, bidders = null) {
+  const id = reqBids.auctionId;
+  const contextBidders = bidders ?? Array.from(new Set(reqBids.adUnits.flatMap(adunit => adunit.bids.map(bid => bid.bidder))))
+  qortexSessionInfo.pageAnalysisdata.contextAdded[id] = contextBidders;
 }
 
 export function setContextData(value) {
-  currentSiteContext = value
+  qortexSessionInfo.currentSiteContext = value
+}
+
+export function setGroupConfigData(value) {
+  qortexSessionInfo.groupConfig = value
+}
+
+/**
+ * Creates page index data for Qortex analysis
+ * @returns {Object} page index object
+ */
+function generateIndexData () {
+  return {
+    pageUrl: document.location.href,
+    title: document.title,
+    text: document.body.textContent.replaceAll(/\r?\n/gi, ' '),
```

Comment:
it seems bizarre to send so much content over the wire, why not send a robot to get copies of pages you don't know the contents of? Also, this prevents anyone from using you from behind a login, as you get the login info